ttnn/cpp/ttnn/operations/transformer/sdpa_decode/device/kernels/compute/sdpa_flash_decode.cpp

```diff
@@ -477,15 +479,38 @@ void MAIN {
     constexpr uint32_t cb_out_l = tt::CB::c_out2;
     constexpr uint32_t cb_out_final = tt::CB::c_out4;
 
-    const uint32_t k_num_chunks = get_arg_val<uint32_t>(0);  // number of chunks in K, where k_num_chunks*Sk_chunk_t = PSt
-    const uint32_t k_chunk_start = get_arg_val<uint32_t>(1);
-    const uint32_t k_chunk_end = get_arg_val<uint32_t>(2);
-    const bool do_reduce = get_arg_val<uint32_t>(3) == 1;
+    const bool do_reduce = get_arg_val<uint32_t>(0) == 1;
+    const uint32_t core_num = get_arg_val<uint32_t>(1);
+    const uint32_t cur_batch = get_arg_val<uint32_t>(2);
+    const uint32_t cur_pos_arg = get_arg_val<uint32_t>(3);
+
     // const uin32_t idle_core = get_arg_val<uint32_t>(4);
 
+    // idle core
+    if (get_arg_val<uint32_t>(0)==65){
+        return;
+    }
+
+    // Get cur_pos
+    uint32_t cur_pos = 0;
+    if (cur_pos_arg!=4294967295){
```

Comment:
would be good to comment on why the hard-coded values are chosen. Maybe define them as constants?

Reply:
end of uint32 range